src/util/TypeTraits.h

```diff
@@ -118,25 +120,54 @@ concept SimilarTo = isSimilar<T, U>;
 
 /// True iff `T` is similar (see above) to any of the `Ts...`.
 template <typename T, typename... Ts>
-concept isTypeAnyOf = (... || isSimilar<T, Ts>);
+concept SimilarToAny = (... || isSimilar<T, Ts>);
 
-/// isTypeContainedIn<T, U> It is true iff type U is a pair, tuple or variant
-/// and T `isSimilar` (see above) to one of the types contained in the tuple,
-/// pair or variant.
+/// True iff `T` is the same as any of the `Ts...`.
 template <typename T, typename... Ts>
-constexpr static bool isTypeContainedIn = false;
+concept SameAsAny = (... || std::same_as<T, Ts>);
 
-template <typename T, typename... Ts>
-constexpr static bool isTypeContainedIn<T, std::tuple<Ts...>> =
-    isTypeAnyOf<T, Ts...>;
-
-template <typename T, typename... Ts>
-constexpr static bool isTypeContainedIn<T, std::variant<Ts...>> =
-    isTypeAnyOf<T, Ts...>;
+namespace detail {
+/*
+The implementation for `SimilarToAnyTypeIn` and `SameAsAnyTypeIn` (see below
+namespace detail). For a call `SimilarToAnyTypeIn<T, U>` , the object would be
+`TypeForSimilarToAnyTypeIn<T>::TemplateType<U>::value`.
+*/
+template <typename T>
+struct TypeForAnyTypeIn {
+  // Structs for `SimilarToAnyTypeIn`.
+  template <typename D>
+  struct SimilarToTemplateType : std::false_type {};
+
+  template <template <typename...> typename Template, typename... Ts>
+  requires SimilarToAny<T, Ts...>
+  struct SimilarToTemplateType<Template<Ts...>> : std::true_type {};
+
+  // Structs for `SameAsAnyTypeIn`.
+  template <typename D>
+  struct SameAsTemplateType : std::false_type {};
+
+  template <template <typename...> typename Template, typename... Ts>
+  requires SameAsAny<T, Ts...>
+  struct SameAsTemplateType<Template<Ts...>> : std::true_type {};
+};
+}  // namespace detail
```

Comment:
This whole impl with the additional outer struct  can actually be made much simpler, [see here](https://godbolt.org/z/v81o89a7z).
You in particular don't need the outer struct which makes everythink much more readable.
Feel free to also change the other implementation in that much simpler way of which I didn't know earlier.

Reply:
Done, see [bdc88dc](https://github.com/ad-freiburg/qlever/pull/1214/commits/bdc88dc9da9a8052b86e18a20f5c2b919e825336).